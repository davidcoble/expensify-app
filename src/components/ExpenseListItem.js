import React from 'react';
import { Link } from 'react-router-dom'


const ExpenseListItem = ({dispatch, description, amount, note, createdAt, id}) => (
  <div>
      <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
      </Link>
      <p>{amount} - {createdAt} - {note}</p>

  </div>
);

export default (ExpenseListItem);