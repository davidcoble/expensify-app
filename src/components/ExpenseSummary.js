import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpenseTotal from '../selectors/expenses-total';

export const ExpenseSummary = (props) => {
    const number = props.expenses.length;
    const totalAmount = selectExpenseTotal(props.expenses);
    return (
        <div>
            {
                number === 0 ? (
                    <p>Viewing no expenses</p>
                ) : (
                    <p>
                        Viewing {number} expense{number > 1 && 's'} totalling {numeral(totalAmount / 100).format('$0,0.00')}.
                    </p>
                )
            }
        </div>
    );
};

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseSummary);
