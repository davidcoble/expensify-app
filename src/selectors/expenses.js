import moment from 'moment';

// Get visible expenses
export default (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
        const createdAtMoment = moment(expense.createdAt);
        const startDateMatch = startDate ? startDate.isSameOrBefore(expense.createdAt) : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(expense.createdAt) : true;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());
        return startDateMatch && endDateMatch && textMatch;
    }).sort((a,b) => {
        if (sortBy === 'date') {
            return a.createdAt < b.createdAt ? 1 : -1;
        } else if (sortBy === 'amount') {
            // console.log("a.amount = " + a.amount);
            // console.log("b.amount = " + b.amount);
            const retval = a.amount < b.amount ? 1 : -1;
            // console.log("retval = " + retval);
            return retval;
        }
    });
};
