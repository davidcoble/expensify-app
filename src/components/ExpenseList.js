import React from 'react';
import {connect} from 'react-redux';
import selectExpenses from '../selectors/expenses';

import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>Please add some Expenses</p>
            ) : (
                props.expenses.map((e) => {
                    return <ExpenseListItem key={e.id} {...e}{...props} />
                })
            )
        }
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
