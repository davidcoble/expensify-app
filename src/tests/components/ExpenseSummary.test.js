import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import expenses from '../fixtures/expenses';

test('should render a Summary with no expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render a Summary with one expense', () => {
    const wrapper = shallow(<ExpenseSummary expenses={[expenses[0]]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should render a Summary with many expenses', () => {
    const wrapper = shallow(<ExpenseSummary expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
});
