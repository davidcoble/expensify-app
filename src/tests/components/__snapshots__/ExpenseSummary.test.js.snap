// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Summary with many expenses 1`] = `
<div>
  <p>
    Viewing 
    3
     expense
    s
     totalling 
    $1,141.95
    .
  </p>
</div>
`;

exports[`should render a Summary with no expenses 1`] = `
<div>
  <p>
    Viewing no expenses
  </p>
</div>
`;

exports[`should render a Summary with one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     expense
     totalling 
    $1.95
    .
  </p>
</div>
`;
